lms/djangoapps/courseware/tests/test_access.py

```diff
@@ -851,7 +802,9 @@ def test_course_catalog_access_num_queries(self, user_attr_name, action, course_
             user = getattr(self, user_attr_name)
             user = User.objects.get(id=user.id)
 
-        if user_attr_name == 'user_staff' and action == 'see_exists' and course_attr_name == 'course_not_started':
```

Comment:
@nasthagiri I think I got this. I have refactored `_has_access_to_course` method. Checking global, staff and instructor access is moved to method `adminstrative_accesses_to_course_for_user` with intention of reusing. Before this `_has_access_to_course` was checking each access (global, staff, instructor) one by one and was returning accordingly. Now, as this all is moved to a method, all accesses are checked at once, resulting into an extra query in one case where course attribute name is `course_default `.